doc/services/device_mgmt/smp_groups/smp_group_0.rst

```diff
@@ -431,7 +431,25 @@ System reset request header fields:
     | ``2``  | ``0``        |  ``5``         |
     +--------+--------------+----------------+
 
-The command sends empty CBOR map as data.
+Normally the command sends empty CBOR map as data, but if previous
+reset attempt has been responded with "rc" code equal ``10`` (busy),
+then following map may be send to force the reset:
+
+.. code-block:: none
+
+    {
+        (opt)"force"       : (int)
+    }
+
+where:
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "force"               | Force reset if value > 0, optional if 0.          |
```

Comment:
Issue with this from a device perspective is it can be used to DoS a device or evade security by having a cheap device which just connects and issues this command over and over again causing the target device to continuously reboot, especially considering if UDP transport is being used then any other device on the network can reboot the device without any security restrictions

Reply:
This should not override restrictions. This is intended, and optional, for resetting device when some SMP process gets stuck and SMP would keep returning busy - this is only to report SMP problem.
Then user is expected to investigate the issue and only force reset when sure that  the risk can be taken, this may take additional steps like allowing reset at all, via other channel. We probably should also add something `MGMT_ERR_NOACCS` or `MGMT_ERR_UNAUTHORIZED` to handle cases like https://github.com/zephyrproject-rtos/zephyr/commit/d67a364ace006f16019dc4fe0bc689df2dea55f8.
Also this will be optional, if implemented for mcumgr lib, but it's main focus is device-to-device update, where one device is in mcuboot.

The SMP does not seem to be designed to safely work with IP stack: UART and BT are the main focus; the comm channel lacks encryption and authorization methods, which means that when you expose IP of the device you may DDoS it several other ways.